### PR TITLE
Render gun cleanup wip

### PR DESCRIPTION
**Pull request by TheWillyJ**
Just some refactoring so this code is remotely legible

### CODE REVIEW

File: src/main/java/com/flansmod/client/model/RenderGun.java [hunk 1]
```diff
@@ -556,332 +551,325 @@ else if (!empty && !animations.reloading)
...
+			renderChargeHandle(model, animations, f);
+			//Render minigun barrels
+			renderMinigunBarrels(type, model, animations, f);
+			// Render the revolver barrel
+			renderRevolverBarrel(model, reloadRotate, f);
+			//Render gun ammo
+			renderAmmo(model, type, gripAttachment, item, empty, animations, reloadRotate, rtype, f, gripItemStack);
+			// Render a static model of the ammo NOT being reloaded
+			//this seems to not be required at all
+			//renderStaticAmmo(type, item, model, f, gripAttachment, gripItemStack);
```
Comment (TheWillyJ) [inline src/main/java/com/flansmod/client/model/RenderGun.java]: I couldn't figure out what this bit was actually doing, it doesn't seem to change anything if it's commented out. Thoughts?

File: src/main/java/com/flansmod/client/model/RenderGun.java [hunk 2]
```diff
@@ -64,32 +61,31 @@ public void renderItem(ItemRenderType type, ItemStack item, Object... data) {
 		// Avoid any broken cases by returning
 		if (!(item.getItem() instanceof ItemGun))
 			return;
-
-		RenderBlocks renderBlocks = (RenderBlocks) data[0];
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/client/model/RenderGun.java:68]: Was this actually doing anything here before? Seems redundant to me?

File: src/main/java/com/flansmod/client/model/RenderGun.java [hunk 3]
```diff
@@ -119,332 +115,272 @@ public void renderOffHandGun(EntityPlayer player, ItemStack offHandItemStack) {
 		if (!offHandGunType.getOneHanded())
 			return;
 
-		renderGun(ItemRenderType.EQUIPPED, offHandItemStack, offHandGunType, animations, true, player);
+		renderGunMovement(ItemRenderType.EQUIPPED, offHandItemStack, offHandGunType, animations, true, player);
 	}
 
-	private void renderGun(ItemRenderType type, ItemStack item, GunType gunType, GunAnimations animations, boolean offHand, Object... data) {
+	private void renderGunMovement(ItemRenderType type, ItemStack item, GunType gunType, GunAnimations animations, boolean offHand, Object... data) {
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/client/model/RenderGun.java]: Could you comment here what renderGunMovement does here?

same with the other renderGun's

Also, would renderGunTransform be better? Or pose? It's like setting up the stance before the actual model's thrown in

File: src/main/java/com/flansmod/client/model/RenderGun.java [hunk 4]
```diff
@@ -119,332 +115,272 @@ public void renderOffHandGun(EntityPlayer player, ItemStack offHandItemStack) {
...
-			switch (type) {
-			case ENTITY: {
-				EntityItem entity = (EntityItem) data[1];
-				GL11.glRotatef(entity.age + (entity.age == 0 ? 0 : smoothing), 0F, 1F, 0F);
-				GL11.glTranslatef(-0.2F + model.itemFrameOffset.x, 0.2F + model.itemFrameOffset.y,
-						0.1F + model.itemFrameOffset.z);
-				break;
+			// Do transforms and rotations depending on how the gun is being rendered/viewed
+			//Ground, item frame, etc
+			if (type.name().equals("ENTITY")) {
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/client/model/RenderGun.java]: Could you check these against the enum instead?

https://lazdude2012.github.io/YATS/enumnet_1_1minecraftforge_1_1client_1_1_i_item_renderer_1_1_item_render_type.html

File: src/main/java/com/flansmod/client/model/RenderGun.java [hunk 5]
```diff
@@ -119,332 +115,272 @@ public void renderOffHandGun(EntityPlayer player, ItemStack offHandItemStack) {
...
+				GL11.glRotatef(60F * reloadRotate, 0F, 0F, 1F);
+				GL11.glRotatef(30F * reloadRotate * flip, 1F, 0F, 0F);
+				GL11.glTranslatef(0.25F * reloadRotate, 0F, 0F);
 				break;
 			}
-			default:
+			//TODO Investigate if all this can be deleted
+			case CUSTOMBOTTOM_CLIP:
+			case CUSTOMPISTOL_CLIP:
+			case CUSTOMSHOTGUN:
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/client/model/RenderGun.java:310]: huh, where these just never implemented? / All have the same behaviour here?

File: src/main/java/com/flansmod/client/model/RenderGun.java [hunk 6]
```diff
@@ -455,94 +391,153 @@ public void renderGun(ItemStack item, GunType type, float f, ModelGun model, Gun
...
-		ItemStack gripItemStack = type.getGripItemStack(item);
-		ItemStack gadgetItemStack = type.getGadgetItemStack(item);
-		ItemStack slideItemStack = type.getSlideItemStack(item);
-		ItemStack pumpItemStack = type.getPumpItemStack(item);
-		ItemStack accessoryItemStack = type.getAccessoryItemStack(item);
-
-		// Gun recoil
 		animations.recoilAmount = model.recoilAmount;
 
-		GL11.glPushMatrix();
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/client/model/RenderGun.java:478]: hm, where's this been moved to?

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 1 above)
Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java]: Looks like it loads ammo models for underbarrels, I didn't know underbarrels could have ammo models!?

**Review by GoldSloth — COMMENTED** [review]
Looks a lot tidier, just a couple things I've mentioned in the comments. Too hard to check everything in the diff, but as long as you've tested stuff it should be fine. Main concern is push/pop I guess?

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 5 above)
Comment (TheWillyJ) [reply src/main/java/com/flansmod/client/model/RenderGun.java:310]: Yeah theyre basically all the same so reduced them to one case

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 2 above)
Comment (TheWillyJ) [reply src/main/java/com/flansmod/client/model/RenderGun.java:68]: Wasnt being used

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 1 above)
Comment (TheWillyJ) [reply src/main/java/com/flansmod/client/model/RenderGun.java]: Ah okay yeah you're probably right

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 2 above)
Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java:68]: Fair enough. Looks like something copied from vanilla code I guess

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 5 above)
Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java:310]: Oke

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 6 above)
Comment (TheWillyJ) [reply src/main/java/com/flansmod/client/model/RenderGun.java:478]: 441-471 wraps this now

Comment (TheWillyJ) [reply src/main/java/com/flansmod/client/model/RenderGun.java:478]: Actually, thats prob not accurate numbers cause my code has further changes, but its good

Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java:478]: Oke

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 1 above)
Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java]: Alr

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 4 above)
Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java]: All good

File: src/main/java/com/flansmod/client/model/RenderGun.java (hunk 3 above)
Comment (GoldSloth) [reply src/main/java/com/flansmod/client/model/RenderGun.java]: Oke, thats good :)

**Comment by GoldSloth** [comment]
I'll merge this without unknown's review since it shouldn't change any behaviour

**Pull request merged**